### PR TITLE
Slightly more explicit coverage for partition-related methods.

### PR DESCRIPTION
**Pull request by tseaver**
Follow-on to #2058.

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes PTAL.

File: gcloud/bigquery/table.py [hunk 1]
```diff
@@ -245,25 +209,22 @@ def partitioning_type(self, value):
         :type value: str
         :param value: partitioning type only "DAY" is currently supported
         """
-        if not (isinstance(value, six.string_types)
-                and value.upper() == "DAY") and value is not None:
+        if value not in ['DAY', None]:
```
Comment (dhermes) [inline gcloud/bigquery/table.py]: Use a tuple, not a list, also might be worth making a constant

File: gcloud/bigquery/table.py [hunk 2]
```diff
@@ -272,13 +233,19 @@ def partition_expiration(self, value):
...
         :param value: partition experiation time in ms
         """
-        if not isinstance(value, int):
-            raise ValueError("must be an integer representing millisseconds")
-        try:
-            self._properties["timePartitioning"]["expirationMs"] = value
-        except KeyError:
-            self._properties['timePartitioning'] = {'type': "DAY"}
-            self._properties["timePartitioning"]["expirationMs"] = value
+        if not isinstance(value, (int, type(None))):
```
Comment (dhermes) [inline gcloud/bigquery/table.py]: Preference for `type(None)` over `types.NoneType`?

File: gcloud/bigquery/table.py [hunk 3]
```diff
@@ -272,13 +233,19 @@ def partition_expiration(self, value):
...
+            raise ValueError(
+                "must be an integer representing millisseconds or None")
+
+        if value is None:
+            if 'timePartitioning' in self._properties:
+                self._properties['timePartitioning'].pop('expirationMs')
+        else:
+            try:
+                self._properties["timePartitioning"]["expirationMs"] = value
```
Comment (dhermes) [inline gcloud/bigquery/table.py]: Single quotes

File: gcloud/bigquery/table.py [hunk 4]
```diff
@@ -272,13 +233,19 @@ def partition_expiration(self, value):
...
+            except KeyError:
+                self._properties['timePartitioning'] = {'type': "DAY"}
+                self._properties["timePartitioning"]["expirationMs"] = value
```
Comment (dhermes) [inline gcloud/bigquery/table.py]: Single quotes (for `"DAY"` on the line above too)

File: gcloud/bigquery/table.py (hunk 2 above)
Comment (tseaver) [inline gcloud/bigquery/table.py]: I didn't want to fuss with the import at the time.

File: gcloud/bigquery/table.py (hunk 1 above)
Comment (tseaver) [inline gcloud/bigquery/table.py]: 384e046

File: gcloud/bigquery/table.py (hunk 2 above)
Comment (tseaver) [inline gcloud/bigquery/table.py]: 384e046

File: gcloud/bigquery/table.py (hunk 3 above)
Comment (tseaver) [inline gcloud/bigquery/table.py]: 384e046

File: gcloud/bigquery/table.py (hunk 4 above)
Comment (tseaver) [inline gcloud/bigquery/table.py]: 384e046

File: gcloud/bigquery/table.py (hunk 2 above)
Comment (tseaver) [inline gcloud/bigquery/table.py]: Py3k doesn't have it!  See https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/150735304#L214

**Comment by dhermes** [comment]
New changes LGTM. I wonder why `types.NoneType` is missing in Py3. Feel free to merge after fixing that attribute error. Squash if you can.

**Comment by dhermes** [comment]
https://bugs.python.org/issue19438

**Comment by tseaver** [comment]
I've rebased to `master` to get the tests fixed (after merging #2065).  I will merge when Travis passes.

**Comment by dhermes** [comment]
@tseaver FYI Travis has passed

**Pull request merged**